dashboard/test/models/concerns/pd/form_test.rb

```diff
@@ -32,6 +62,16 @@ class Pd::FormTest < ActiveSupport::TestCase
     assert_equal ["can't be blank"], form.errors.messages[:form_data]
   end
 
+  test 'pd form only validates form data when changed' do
+    form = DummyFormWithRequiredFields.new
+    form.form_data = {firstField: "foo"}.to_json
+    form.save
+
+    assert form.valid?
+    form.form_data = {firstField: "bar"}.to_json
+    assert_equal false, form.valid?
+  end
+
```

Comment:
See above. I think it might be worth (and simpler) persisting to the (test) db. Otherwise yes please add comments.